docs/admin/postgres12_end_of_life_notice.mdx

```diff
@@ -0,0 +1,45 @@
+# Postgres 12 End of Life
+
+Sourcegraph's required minimum version of Postgres will be updated to Postgres 16 with upcoming releases, Sourcegraph 5.10. This change in required minimum version is a result of Postgres 12, the previous required minimum version, reaching end of life. This notification is coming to you because you are running a self-hosted instance of Sourcegraph. Please review the below information and reach out if you have any questions or would like to review the impact this change will have on you!
+
+## What
+Currently, Sourcegraph's minimum required version of Postgres, the database management system used for all Sourcegraph instances, is Postgres 12. Postgres 12 is reaching end-of-life and will no longer be supported by Postgres. Because of this, Sourcegraph will be updating our minimum supported version to Postgres 16.
+
+## Impact
+The impact to you depends on how you currently manage your Sourcegraph databases. Please first determine which of the following two buckets you fall into to determine what impact this change will have on your instance. If you are uncertain of which bucket you fall under, please check your deployment configuration or reach out to your account team.
+
+* **Bucket 1 (Built-in Databases)** - Your Sourcegraph databases are deployed using the built-in Postgres images standardly provided by Sourcegraph
+* **Bucket 2 (External Databases)** - Your Sourcegraph databases run in an externally managed database service separate from your Sourcegraph deployment, such as AWS RDS or Google CloudSQL (External Database)
+
+### Built-in Databases
+If you are using Sourcegraph's released Postgres images, these images will be updated to Postgres 16 as part of the Sourcegraph 5.10 release (expected for November 25th, 2024) and all near term subsequent releases. With this update, Sourcegraph will experience a longer-than-average downtime at the time of the upgrade to version 5.10 or beyond. This downtime is proportional to the amount of code indexed in your instance, and can range from approximately 15 minutes to a couple of hours. The downtime will only occur at the first upgrade to version 5.10 or beyond, and not during any subsequent upgrades. 
+
+We strongly recommend that you consider this downtime prior to upgrading to ensure minimal disruption to your users. We also strongly recommend creating a backup of your database before the upgrade process.
+
+### External Databases
+If you are using externally managed databases for your Sourcegraph instance, you will need to manually upgrade the version of Postgres you are using to Postgres 16. For Sourcegraph version 5.12 (expected for release on January 15, 2025) and subsequent versions, Sourcegraph will no longer run on any version of Postgres lower than Postgres 16, and you will not be able to upgrade your instance to any of those Sourcegraph versions unless that minimum required version of Postgres is met.
```

Comment:
>For Sourcegraph version 5.12

This is actually going to be Sourcegraph 6.0 from my understanding, however we should triple check with marketing first.

Reply:
Maybe we just say, 'the January 15, 2025 Sourcegraph release'?